File: operation-spring/entity/dto/CostElementOperation.java
package com.prompt.operation.entity.dto;

import com.prompt.operation.core.dto.OperationDto;

import java.time.LocalDateTime;

public class CostElementOperation extends OperationDto {

    private String name;

    private String symbol;

    private byte[] totalBudget;

    public CostElementOperation() {
    }

    public CostElementOperation(LocalDateTime occurredAt, String name, String symbol, byte[] totalBudget) {
        super(occurredAt);
        this.name = name;
        this.symbol = symbol;
        this.totalBudget = totalBudget;
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public byte[] getTotalBudget() {
        return totalBudget;
    }

    public void setTotalBudget(byte[] totalBudget) {
        this.totalBudget = totalBudget;
    }
}

File: operation-nest/entity/cost-element-operation/cost-element-operation.ts
import { IsNotEmpty, MaxLength } from 'class-validator';
import { OperationDto } from 'src/operation/core/domain/operation.dto';
import { IsEncryptedNumber } from '../../../cryptography/is-encrypted-number';

export class CostElementOperation extends OperationDto {

    @IsNotEmpty()
    name: string;

    @IsNotEmpty()
    @MaxLength(3)
    symbol: string;

    @IsNotEmpty()
    @IsEncryptedNumber()
    totalBudget: string;

    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    constructor(data?: any) {
        super(data);
        this.name = data && data.name || null;
        this.symbol = data ? data.symbol : null;
        this.totalBudget = data && data.totalBudget || null;
    }
}
